lib/Dialect/FIRRTL/FIRRTLOpInterfaces.cpp

```diff
@@ -130,6 +130,23 @@ LogicalResult circt::firrtl::verifyModuleLikeOpInterface(FModuleLike module) {
   if (failed(verifyNoInputProbes(module)))
     return failure();
 
+  // Enforce public and ext module requirements.
+  if (module.isPublic() || isa<FExtModuleOp>(module)) {
+    for (auto [idx, type] : llvm::enumerate(
+             module.getPortTypesAttr().getAsValueRange<TypeAttr>())) {
+      if (auto ftype = dyn_cast<FIRRTLType>(type)) {
+        if (ftype.hasUninferredWidth())
+          return emitError(module.getPortLocation(idx),
+                           "public or ext module port must have known width");
+        // Presently abstract reset on extmodule is not explicitly disallowed in
```

Comment:
That seems like a deficiency in the spec, not the intended behavior.  I would be ok with updating the spec and enforcing this.

Reply:
https://github.com/chipsalliance/firrtl-spec/pull/181

Reply:
Adding the checks to the parse per current spec behavior, will extend once that is made illegal there.